HRIS.Services/Services/FieldCodeOptionsService.cs

```diff
@@ -83,15 +109,19 @@ public async Task<List<FieldCodeOptionsDto>> UpdateFieldCodeOptions(List<FieldCo
             if (!check) _ = await _db.FieldCodeOptions.Delete(option.Id);
         }
 
-        var updatedFieldCodeOptions = await GetFieldCodeOptions(fieldCodeOptionsDto[0].FieldCodeId);
+        var updatedFieldCodeOptions = await GetFieldCodeOptionsById(fieldCodeOptionsDto[0].FieldCodeId);
         return updatedFieldCodeOptions;
     }
 
     public async Task<FieldCodeOptionsDto> DeleteFieldCodeOptions(FieldCodeOptionsDto fieldCodeOptionsDto)
     {
-        var ifFieldCodeOption = await GetFieldCodeOptions(fieldCodeOptionsDto.Id);
-        if (ifFieldCodeOption == null)
-            throw new CustomException("No field with that name found");
+        var fieldCodeOptionExist = await FieldCodeOptionsExists(fieldCodeOptionsDto.Id);
+
+        if (!fieldCodeOptionExist)
+            throw new CustomException("Field Code Option does not exist");
+
+        if (_identity.IsSupport == false && fieldCodeOptionsDto.Id != _identity.EmployeeId)
```

Comment:
`fieldCodeOptionsDto.Id != _identity.EmployeeId`

Is `fieldCodeOptionsDto.Id` the `EmployeeId` ?